src/views/MarketPlace/components/AdminPage/index.jsx

```diff
@@ -63,13 +68,71 @@ const AdminMarketplaceThreads = () => {
     fetchThreads();
   }, [page, rowsPerPage]);
 
+  useEffect(() => {
+    const trySearchById = async () => {
+      if (!searchTrigger) return;
+
+      setLoading(true);
+      try {
+        const pageSize = 100;
+        let currentPage = 1;
+        let hasMore = true;
+
+        while (hasMore) {
+          const threadsPage = await marketplaceService.getAdminThreads({
+            page: currentPage,
+            pageSize,
+          });
+
+          for (const thread of threadsPage) {
+            const history = await marketplaceService.getThreadEditHistory(thread.ThreadId);
+            const match = history.find((h) => h.Id.toString() === searchTrigger);
+            if (match) {
+              setEditHistory(history);
+              setIsModalOpen(true);
+              setSearchQuery('');
+              setSearchTrigger('');
+              setLoading(false);
+              return;
+            }
+          }
+
+          hasMore = threadsPage.length === pageSize;
+          currentPage++;
+        }
+
+        alert(`No thread history found with ID ${searchTrigger}`);
+      } catch (err) {
+        console.error('Error searching thread history:', err);
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    trySearchById();
+  }, [searchTrigger]);
+
   if (loading) return <CircularProgress />;
 
   return (
     <Paper>
+      <Box p={2}>
+        <TextField
+          label="Search ID Number"
+          variant="outlined"
+          value={searchQuery}
+          onChange={(e) => setSearchQuery(e.target.value)}
+          onKeyDown={(e) => {
+            if (e.key === 'Enter') {
+              setSearchTrigger(searchQuery.trim());
+            }
+          }}
```

Comment:
There needs to be some way for a mouse user to trigger the search, and preferably a visible, obvious way to trigger the search because not every user will think to hit enter.

Also, since you're only setting `searchTrigger` in response to this `onKeyDown`, you should just make `trySearchById` a regular function and directly call it when the event occurs, rather than setting an intermediate state:
```suggestion
          onKeyDown={(e) => {
            if (e.key === 'Enter') {
              trySearchById(searchQuery.trim());
            }
          }}
```

Reply:
Fixed!!